internal/stacks/stackruntime/internal/stackeval/component_config.go

```diff
@@ -414,6 +414,19 @@ func (c *ComponentConfig) neededProviderSchemas(ctx context.Context, phase EvalP
 		if pTy == nil {
 			continue // not our job to report a missing provider
 		}
+
+		// If this phase has a dependency lockfile, check if the provider is in it.
+		depLocks := c.main.DependencyLocks(phase)
+		if depLocks != nil {
+			// Check if the provid
+			providerLockfileDiags := CheckProviderInLockfile(*depLocks, pTy, decl.DeclRange)
+
+			// We report these diagnostics in a different place
+			if providerLockfileDiags.HasErrors() {
+				return providerSchemas, diags, true
+			}
+		}
```

Comment:
bit of a nit: I guess we need to check the lock file since we get some crazy error later in this function if a provider is missing Could you add a comment just briefly explaining why we need to check the lock file at all?

Also, I see in the other implementation of this within ComponentInstance, that it doesn't return early at all and simply skips the problematic provider. Could the comment you add explain the need for the difference in these two approaches?